tensorflow_probability/python/mcmc/transformed_kernel_test.py

```diff
@@ -308,6 +328,89 @@ def test_is_calibrated(self):
             FakeInnerKernel(lambda x: -x**2 / 2, False),
             tfb.Identity()).is_calibrated)
 
+  def test_transform_then_step(self):
+    new_kernel = transform_then_step(tfb.Identity())
+    pkr_one, pkr_two = self.evaluate([
+        new_kernel.bootstrap_results(9.),
+        new_kernel.bootstrap_results(5.)
+    ])
+    self.assertNear(9., pkr_one.inner_results.transformed_state, err=1e-6)
+    self.assertNear(5., pkr_two.inner_results.transformed_state, err=1e-6)
+
+  def test_step_then_transform(self):
+    new_kernel = step_then_transform(tfb.Identity())
+    pkr_one, pkr_two = self.evaluate([
+        new_kernel.bootstrap_results(9.),
+        new_kernel.bootstrap_results(5.),
+    ])
+    self.assertNear(9., pkr_one.transformed_state, err=1e-6)
+    self.assertNear(5., pkr_two.transformed_state, err=1e-6)
+
+  def test_bijector_valid_transform_then_step(self):
+    new_kernel = transform_then_step(tfb.Exp())
+    pkr_one, pkr_two = self.evaluate([
+        new_kernel.bootstrap_results(2.),
+        new_kernel.bootstrap_results(9.),
+    ])
+    self.assertNear(np.log(2.),
+                    pkr_one.inner_results.transformed_state,
+                    err=1e-6)
+    self.assertNear(np.log(9.),
+                    pkr_two.inner_results.transformed_state,
+                    err=1e-6)
+
+  def test_bijector_valid_step_then_transform(self):
+    new_kernel = step_then_transform(tfb.Exp())
+    pkr_one, pkr_two = self.evaluate([
+        new_kernel.bootstrap_results(2.),
+        new_kernel.bootstrap_results(9.),
+    ])
+    self.assertNear(np.log(2.), pkr_one.transformed_state, err=1e-6)
+    self.assertNear(np.log(9.), pkr_two.transformed_state, err=1e-6)
+
+  def test_step_size_changed(self):
+    target_dist = tfd.MultivariateNormalDiag(loc=[0., 0.], scale_diag=[1., 10.])
+    # `hmc_kernel`'s step size is far from optimal
+    hmc_kernel = tfp.mcmc.HamiltonianMonteCarlo(
+        target_log_prob_fn=target_dist.log_prob,
+        num_leapfrog_steps=27,
+        step_size=10)
+    step_adaptation_kernel = tfp.mcmc.SimpleStepSizeAdaptation(
+        inner_kernel=hmc_kernel,
+        num_adaptation_steps=9)
+    trans_kernel = tfp.mcmc.TransformedTransitionKernel(
+        inner_kernel=step_adaptation_kernel,
+        bijector=tfb.Exp()
+    )
+    kernel_results = trans_kernel.inner_kernel.bootstrap_results(tf.zeros(2))
+    for _ in range(2):
+      _, kernel_results = trans_kernel.inner_kernel.one_step(tf.zeros(2),
+                                                             kernel_results)
+    adapted_step_size = self.evaluate(
+        kernel_results.inner_results.accepted_results.step_size)
+    # the step size for `FakeInnerKernel` is 10
+    self.assertFalse(adapted_step_size == 10)
```

Comment:
It would be better to check that it moved a significant amount, such as self.assertLess(adapted_step_size, 8).  Take more transitions if necessary.

Also, this test is randomized, so it should follow our test reproducibility conventions:
- Create an initial seed stream using `strm = test_util.test_seed_stream()`
- Pass explicit seeds to all random functions as `foo(..., seed=strm())`
- Check that you got it right by inserting a debug print somewhere and seeing that the adapted step size you get is the same across test runs
- Check that your test bound is reasonable by making sure the test passes consistently if the seed is varied.  You check this by running the test with `bazel test :foo --test_arg=--vary_seed --runs_per_test=10` (or 100, if the test is fast enough for that to be reasonable).

Reply:
Done with similar flavor to the rest of the file (but I did test it out with multiple seeds using the test seed stream)